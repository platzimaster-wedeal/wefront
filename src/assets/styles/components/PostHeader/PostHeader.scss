@import '../../partials/colors.scss';
@import '../../partials/sizes.scss';
@import '../../Breakpoints.scss';

.post-header {
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  width: 100%;
  height: auto;
  max-height: 240px;
  padding: $space-size-extra-small 0;

  &__picture {
    margin: 0 $space-size-normal; 
    cursor: pointer;
  }

  & .image {
    width: 42px;
    height: 42px;

    @include media(tablet) {
      width: 70px;
      height: 70px;

    }
  }

  &__user-info {
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    width: 70%;

    &--main {
      display: flex;
      flex-flow: column nowrap;
      margin-bottom: 2px;

      @include media(tablet) {      
        flex-flow: row nowrap;
        align-items: flex-end;

      }

      & span {
        padding-top: $space-size-extra-small;

        @include media(tablet){
          padding: 0;
        }
      }
    }

    

    h3 {
      color: $color-primary;
      @include font-size($font-size-normal);
      cursor: pointer;
      
      @include media(tablet) {
        @include font-size($font-size-medium + 3);
      }
    }

    strong {
      display: none;
      font-family: 'WorkSansExtraLight';
      font-style: italic;
      @include font-size($font-size-small);
      color: $color-black-60;
      letter-spacing: .6px;
      padding: 6px 0;

      @include media(tablet) {
        display: inline-block;
      }
    }
  }

  &__time {
    @include font-size($font-size-small);
    @include media(tablet) {
      @include font-size($font-size-normal);
    }
  }
}
