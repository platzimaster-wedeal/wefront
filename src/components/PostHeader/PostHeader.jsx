import React from "react";
import { useHistory } from "react-router-dom";


// Stlyes
import "../../assets/styles/components/PostHeader/PostHeader.scss";

// Components
import ProfilePicture from "../ProfilePictures/ProfilePicture";
import Labels from "../Labels/Labels";

const PostHeader = ({ id, name, share, time, profession, picture }) => {
 const history = useHistory();
 const DEFAULT_IMG = 'https://imgurl.me/images/2020/09/11/profilededaultbb7053428141edf1.png'
 const goToProfile = () => history.push(`/user/profile/${id}`);

 return (
  <section className="post-header">
   <figure className="post-header__picture" onClick={goToProfile}>
    <ProfilePicture active source={picture || DEFAULT_IMG } alt={name} />
   </figure>
   <div className="post-header__user-info">
    <div className="post-header__user-info--main">
     <h3 onClick={goToProfile}>{name || "Alan Walks"}</h3>
     <span>{share ? "Has shared!" : "Has a new problem!"}</span>
    </div>
    <strong className="post-header__user-info--profession">
     {share && profession && "Engineer"}
     {!share && <Labels name="Engineer" />}
    </strong>
   </div>
   <small className="post-header__time">{time || "1hr"}</small>
  </section>
 );
};

export default PostHeader;
