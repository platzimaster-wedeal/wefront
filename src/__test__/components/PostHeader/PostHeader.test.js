import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import PostHeader from "../../../components/PostHeader/PostHeader";

describe("Test for the component PostHeader :D", () => {
 const PostHeaderMounted = mount(<PostHeader />);
 const PostHeaderCreated = create(<PostHeader />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(PostHeaderCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(PostHeaderMounted.length).toEqual(1);
  });
  test("Should the name of the User be a string", () => {
   expect(
    typeof PostHeaderMounted.find(".post-header__user-info--main h3").text()
   ).toEqual("string");
  });

  test("Should the type of post be a string", () => {
   expect(
    typeof PostHeaderMounted.find(".post-header__user-info--main span").text()
   ).toEqual("string");
  });
  test("Should be the type of post be equal to Has a new problem!", () => {
   expect(
    PostHeaderMounted.find(".post-header__user-info--main span").text()
   ).toEqual("Has a new problem!");
  });
  test("should be the type of post equal to Has shared!", () => {
   expect(
    mount(<PostHeader share />)
     .find(".post-header__user-info--main span")
     .text()
   ).toEqual("Has shared!");
  });

  test("should the time of the post be a number", () => {
   expect(
    typeof Number(PostHeaderMounted.find(".post-header__time").text().charAt(0))
   ).toEqual("number");
  });
 });
});
