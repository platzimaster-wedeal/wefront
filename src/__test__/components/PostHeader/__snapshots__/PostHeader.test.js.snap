// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component PostHeader :D Visual tests for the component should the comopnent have the same tree 1`] = `
<section
  className="post-header"
>
  <figure
    className="post-header__picture"
    onClick={[Function]}
  >
    <img
      className="image image__isOnline undefined"
      src={Object {}}
    />
  </figure>
  <div
    className="post-header__user-info"
  >
    <div
      className="post-header__user-info--main"
    >
      <h3
        onClick={[Function]}
      >
        Alan Walks
      </h3>
      <span>
        Has a new problem!
      </span>
    </div>
    <strong
      className="post-header__user-info--profession"
    >
      <span
        className="label label__inactive"
      >
        Engineer
      </span>
    </strong>
  </div>
  <small
    className="post-header__time"
  >
    1hr
  </small>
</section>
`;
